external-tracer/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+//! This module generates traces by connecting to an external tracer
+
+use eth_types::evm_types::Gas;
+use eth_types::geth_types::{
+    Account, BlockConstants, GethData, Transaction, MOCK_CHAIN_ID,
+    MOCK_COINBASE,
+};
+use eth_types::{
+    self, fix_geth_trace_memory_size, Address, Block, Bytes, ChainConstants,
+    Error, GethExecStep, GethExecTrace, Hash, Word, U64,
+};
+use serde::Serialize;
+
+/// Create a new block with a single tx that executes the code found in the
+/// account with address 0x0 (which can call code in the other accounts),
+/// with the given gas limit.
+/// The trace will be generated automatically with the external_tracer
+/// from the accounts code.
+pub fn create_tx_by_accounts(
```

Comment:
can we move `create_tx_by_accounts` `create_tx_by_steps` `new_tx` `new_block` and `new_chain_constants` to `mock`?

because they seem mock-specific.

and in `mock::create_tx_by_accounts` we can call `external_tracer::trace`